src/transaction/TransactionController.ts

```diff
@@ -1273,6 +1241,68 @@ export class TransactionController extends BaseController<
     );
   }
 
+  /**
+   * Method to verify the state of a transaction using the Blockchain as a source of truth
+   * @param meta Local transaction to verify data in blockchain
+   * @returns Promise with [TransactionMeta, boolean]
+   */
+  private async blockchainTransactionStateReconciler(
+    meta: TransactionMeta,
+  ): Promise<[TransactionMeta, boolean]> {
+    const { status, transactionHash } = meta;
+    switch (status) {
+      case TransactionStatus.confirmed:
+        const txReceipt = await query(this.ethQuery, 'getTransactionReceipt', [
+          transactionHash,
+        ]);
+
+        if (!txReceipt) {
+          return [meta, false];
+        }
+
+        meta.verifiedOnBlockchain = true;
+        meta.transaction.gasUsed = txReceipt.gasUsed;
+        const txReceiptStatus = Boolean(Number(txReceipt.status));
```

Comment:
Nit: it might be simpler to compare to zero directly here, if that's the intent.

Reply:
Fixed!